.github/workflows/run-test-sh.yml

```diff
@@ -0,0 +1,39 @@
+name: run-test-shell
+
+on:
+  pull_request:
+    branches:
+      - master
+      - stable
+
+  push:
+    branches:
+      - master
+      - stable
+
+jobs:
+  main:
+    name: Build and test image
+    strategy:
+      fail-fast: false # Finish other jobs on failure
+      matrix:
+        include:
+          # - { version: dmd-nightly, exec: dmd   }
+          - { version: dmd-beta,    exec: dmd   }
+          - { version: dmd,         exec: dmd   }
+          - { version: ldc-beta,    exec: ldmd2 }
+          - { version: ldc,         exec: ldmd2 }
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: "Build image"
+        run: docker build -t "dlangtour/core-exec:${{ matrix.version }}" --build-arg "DLANG_VERSION=${{ matrix.version }}" --build-arg "DLANG_EXEC=${{ matrix.exec }}" .
```

Comment:
Instead of doing it manually, use the docker action.
If you want an example, we do this:
https://github.com/bpfkorea/agora/blob/e5fc8fcd925c81c5fcff354880868b5fdbeffc5b/.github/workflows/agora-docker.yml